File: shoring/compat/src/main/java/com/panopset/compat/Listop.java
package com.panopset.compat;

public class Listop {
    public String filter(String input, String filter) {


        if (input == null) {
            Logop.warn("No input.");
            return "";
        }

        if (filter == null) {
            Logop.warn("No input.");
            return "";
        }

        StringBuffer sb = new StringBuffer();
        for (String  inp :  Stringop.stringToList(input)) {
            for (String fs : Stringop.stringToList(filter)) {
                if (inp.equals(fs)) {
                    continue;
                }
                sb.append(inp);
                sb.append(Stringop.getEol());
            }
        }
        return sb.toString();
    }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/Listop.kt
package com.panopset.compat

class Listop {
    fun filter(input: String, filter: String): String {
        val sb = StringBuffer()
        FOR@ for (inp in Stringop.stringToList( input)) {
            for (fs in Stringop.stringToList( filter)) {
                if (inp == fs) {
                    continue@FOR
                }
            }
            sb.append(inp)
            sb.append(Stringop.getEol())
        }
        return sb.toString()
    }
}
